torch/csrc/jit/script/sugared_value.cpp

```diff
@@ -280,6 +281,8 @@ Value* SimpleValue::getelem(const SourceRange&loc, Function& m, Value* i) {
   Value* cur_elem = nullptr;
   if (val_type->cast<ListType>()) {
     cur_elem = g.insert(aten::select, {val, i}, {}, loc);
+  } else if (val_type->cast<StringType>()) {
```

Comment:
It would be nice to standardize a builtin like "__getelem__" to be used here rather than have a big if branch that emits different code, similar to our other magic methods. Furthermore, we should make serialization emit this as `foo[i]` so that we do not end up getting committed to preserving these operator names.

Reply:
I agree with this, will make a separate PR to define it. We will need to follow up on design the deprecation strategy of those prim operators that should be deleted.